src/cheri_insts_end.sail

```diff
@@ -0,0 +1,75 @@
+/*  (ERC) under the European Union’s Horizon 2020 research and innovation                */
+/*  programme (grant agreement 789108, ELVER).                                           */
+/*                                                                                       */
+/*  Redistribution and use in source and binary forms, with or without                   */
+/*  modification, are permitted provided that the following conditions                   */
+/*  are met:                                                                             */
+/*  1. Redistributions of source code must retain the above copyright                    */
+/*     notice, this list of conditions and the following disclaimer.                     */
+/*  2. Redistributions in binary form must reproduce the above copyright                 */
+/*     notice, this list of conditions and the following disclaimer in                   */
+/*     the documentation and/or other materials provided with the                        */
+/*     distribution.                                                                     */
+/*                                                                                       */
+/*  THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS''                   */
+/*  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED                    */
+/*  TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A                      */
+/*  PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR                  */
+/*  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,                         */
+/*  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT                     */
+/*  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF                     */
+/*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND                  */
+/*  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,                   */
+/*  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT                   */
+/*  OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF                   */
+/*  SUCH DAMAGE.                                                                         */
+/*=======================================================================================*/
+
+/* Put the illegal instructions last to use their wildcard match. */
+
+/* Wilcard match for 32-bit non-capmode encodings. */
+union clause ast = NOT_CAPMODE : word
+function clause execute NOT_CAPMODE(s) = { internal_error("should be unreachable"); RETIRE_FAIL }
+mapping clause encdec_capmode = NOT_CAPMODE(s) <-> s
+end encdec_capmode
+
+/* Wilcard match for 16-bit non-capmode encodings. */
+union clause ast = NOT_C_CAPMODE : half
+function clause execute NOT_C_CAPMODE(s) = { internal_error("should be unreachable"); RETIRE_FAIL }
```

Comment:
Same for this error. Just thinking of similar feedback on PR in sail-riscv repo.
```suggestion
function clause execute NOT_C_CAPMODE(s) = { internal_error("should be unreachable execute(NOT_C_CAPMODE(" ^ BitStr(s) ^ "))"); RETIRE_FAIL }
```

Reply:
LGTM